### PR TITLE
QA: bump php and phpuinit for the next major

### PR DESCRIPTION
**Pull request by Xerkus**
|    Q          |   A
|-------------- | ------
| Documentation |no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description
Next branch is a baseline for the next major development and would constitute a series of changes before it is ready to be brought into develop. It was easier to drop the next branch since the only merged PR zendframework/zend-mvc#301 was bumping minimum PHP, prior to Laminas migration.

This PR bumps minimum PHP version to 7.3 and updates phpunit to latest with related fixes to keep suite running but skips cleanup for consistency or quality as much of the existing tests will need to be reworked or even outright dropped. Keeping changes to the minimum would simplify forward merge of 3.x releases.

Replace section for zend-mvc is also dropped as it is not applicable for major releases.

### CODE REVIEW

File: test/Service/ServiceListenerFactoryTest.php [hunk 1]
```diff
@@ -23,25 +24,23 @@ public function setUp()
...
     }
 
-    /**
-     * @expectedException        Laminas\ServiceManager\Exception\ServiceNotCreatedException
-     * @expectedExceptionMessage The value of service_listener_options must be an array, string given.
-     */
     public function testInvalidOptionType()
     {
+        $this->expectException(ServiceNotCreatedException::class);
+        $this->expectExceptionMessage('The value of service_listener_options must be an array, string given.');
```
Comment (michalbundyra) [inline test/Service/ServiceListenerFactoryTest.php:30]: these lines should be just before call, as it is now always clear where the exception is thrown

here, and all places where expectException has been added below.

**Review by michalbundyra — COMMENTED** [review]
@Xerkus so far it looks fine for me.

For the next major, maybe we should consider PHP 7.4 as we should support previous major anyway for some time. Tbh I forgot for how long.

File: test/Service/ServiceListenerFactoryTest.php (hunk 1 above)
Comment (Xerkus) [reply test/Service/ServiceListenerFactoryTest.php:30]: It was automated fix by phpstorm, I think.
I do not worry about that too much at this point: I intend to do a massive tests refactoring following changes to be introduced across the board for the next major and in order to improve tests quality.

I'll push update shortly

**Comment by Xerkus** [comment]
> For the next major, maybe we should consider PHP 7.4

There are no 7.4 features that are currently needed, i think. That can be revisited in 4-5 months when 7.3 goes into security only.

**Comment by michalbundyra** [comment]
@Xerkus 

> There are no 7.4 features that are currently needed

Have you considered adding property typehints? It simplifies a lot, imo :)

**Comment by Xerkus** [comment]
@michalbundyra that feature comes with a performance hit. I wouldn't be comfortable using it without a decent benchmark.

**Comment by michalbundyra** [comment]
@Xerkus 

for ref: https://wiki.php.net/rfc/typed_properties_v2#performance

you can wait for a benchmarks, but in general advantage of using it is quite big, imo :) and it looks like, that the total impact is very low.

**Comment by Xerkus** [comment]
@michalbundyra nice. I remember early assessments that suggested bigger impact.
I will make use of those on the next major stage, the one that should switch to native PSR-7

**Pull request merged**